src/Evolution/Systems/GrMhd/ValenciaDivClean/ConservativeFromPrimitive.hpp

```diff
@@ -0,0 +1,60 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+
+class DataVector;
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace grmhd {
+namespace ValenciaDivClean {
+
+/*!
+ * \brief Compute the conservative variables from primitive variables
+ *
+ * \f{align*}
+ * {\tilde D} = & \sqrt{\gamma} \rho W \\
+ * {\tilde S}_i = & \sqrt{\gamma} \left( \rho h W^2 v_i + B^m B_m v_i - B^m v_m
+ * B_i \right) \\
+ * {\tilde \tau} = & \sqrt{\gamma} \left[ \rho h W^2 - p - \rho W - \frac{1}{2}
+ * B^m v_m + \frac{1}{2} B^m B_m \left( 1 + v^m v_m \right) \right] \\
+ * {\tilde B}^i = & \sqrt{\gamma} B^i \\
+ * {\tilde \Phi} = & \sqrt{\gamma} \Phi
+ * \f}
+ * where the conserved variables
```

Comment:
adding a blank comment line above will prevent clang-format from doing silly things in the future :)